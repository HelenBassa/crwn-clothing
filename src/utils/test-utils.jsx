import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../store/store";


afterAll(() => jest.clearAllMocks());

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';
export * from '@testing-library/user-event';

export { customRender as render };