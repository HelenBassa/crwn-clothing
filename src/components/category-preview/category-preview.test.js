import { render } from "../../utils/test-utils";
import CategoryPreview from "./category-preview.component";

test("should render CategoryPreview component", () => {
  const mockTitle = "MockTitle";
  const mockProducts = ["Hats", "Jeans", "Skirts", "Shoes"];

  const { container } = render(
    <CategoryPreview title={mockTitle} products={mockProducts} />
  );
  expect(container).toMatchSnapshot();
});
