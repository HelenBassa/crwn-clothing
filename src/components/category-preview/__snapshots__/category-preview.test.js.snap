// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CategoryPreview component 1`] = `
<div>
  <div
    class="sc-jrcTuL eUAPUz"
  >
    <h2>
      <a
        class="sc-kDvujY gULvDP"
        href="/MockTitle"
      >
        MOCKTITLE
      </a>
    </h2>
    <div
      class="sc-ipEyDJ gaiQaj"
    >
      <div
        class="sc-gKPRtg gUrvDi"
      >
        <img
          alt="undefined"
        />
        <div
          class="sc-iBYQkv hzJupy"
        >
          <span
            class="sc-ftTHYK hBCWDW"
          />
          <span
            class="sc-pyfCe eraeQQ"
          />
        </div>
        <button
          class="sc-dkrFOg sc-eDvSVe KWCcW dVmvZe"
        >
          Add to cart
        </button>
      </div>
      <div
        class="sc-gKPRtg gUrvDi"
      >
        <img
          alt="undefined"
        />
        <div
          class="sc-iBYQkv hzJupy"
        >
          <span
            class="sc-ftTHYK hBCWDW"
          />
          <span
            class="sc-pyfCe eraeQQ"
          />
        </div>
        <button
          class="sc-dkrFOg sc-eDvSVe KWCcW dVmvZe"
        >
          Add to cart
        </button>
      </div>
      <div
        class="sc-gKPRtg gUrvDi"
      >
        <img
          alt="undefined"
        />
        <div
          class="sc-iBYQkv hzJupy"
        >
          <span
            class="sc-ftTHYK hBCWDW"
          />
          <span
            class="sc-pyfCe eraeQQ"
          />
        </div>
        <button
          class="sc-dkrFOg sc-eDvSVe KWCcW dVmvZe"
        >
          Add to cart
        </button>
      </div>
      <div
        class="sc-gKPRtg gUrvDi"
      >
        <img
          alt="undefined"
        />
        <div
          class="sc-iBYQkv hzJupy"
        >
          <span
            class="sc-ftTHYK hBCWDW"
          />
          <span
            class="sc-pyfCe eraeQQ"
          />
        </div>
        <button
          class="sc-dkrFOg sc-eDvSVe KWCcW dVmvZe"
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
</div>
`;
